quelpa-use-package.el

```diff
@@ -37,17 +37,35 @@
 
 ;;; Code:
 
-(defvar quelpa-use-package-inhibit-loading-quelpa nil
+(defgroup quelpa-use-package nil
+  "Quelpa handler for `use-package'."
+  :prefix "quelpa-use-package"
+  :group 'use-package)
+
+(defcustom quelpa-use-package-inhibit-loading-quelpa nil
   "If non-nil, `quelpa-use-package' will do its best to avoid
 loading `quelpa' unless necessary. This improves performance, but
-can prevent packages from being updated automatically.")
+can prevent packages from being updated automatically."
+  :type 'boolean
+  :group 'quelpa-use-package)
+
+(defcustom quelpa-use-package-as-source-of-truth nil
+  "If non-nil, only packages installed via this will be managed by `quelpa'.
+This option will disable `quelpa-persistent-cache-p'."
+  :type 'boolean
+  :group 'quelpa-use-package)
 
 (require 'cl-lib)
 (unless quelpa-use-package-inhibit-loading-quelpa
   (require 'quelpa))
 (require 'use-package-core)
 
 (defvar quelpa-use-package-keyword :quelpa)
+(defvar quelpa-cache)
+(defvar quelpa-persistent-cache-p)
+
+(when quelpa-use-package-as-source-of-truth
+  (setq quelpa-persistent-cache-p nil))
```

Comment:
This form belongs in the `defcustom`'s setter so it will take effect when the option is changed.